### PR TITLE
Feature/search improvement

### LINKED ISSUE #20: Allow URL query parameters
Allow users to make queries using basic URL query parameters, e.g. http://localhost:5000/name?last=Aakervik, while maintaining the previous URL parsing functionality, e.g. http://localhost:5000/badge-lookup/4810.

This is in pursuit of allowing for more extensible searching/querying (issues 17 and 19)

### PR DESCRIPTION
**Pull request by mpuckett159**
Closes #20

Two main commits here, one to enable query params, second to cut the HTML code over to make use of them.

### CODE REVIEW

File: src/views/index.html [hunk 1]
```diff
@@ -46,21 +46,5 @@ <h1>
     <a href="http://twitter.com/AetherUnbound">If I break message my mom</a>
 </footer>
 
-
-<!-- Your web-app is https, so your scripts need to be too -->
-<script src="https://code.jquery.com/jquery-2.2.1.min.js"
```
Comment (AetherUnbound) [inline src/views/index.html:51]: Yay, less javascript!

File: src/app.py [hunk 2]
```diff
@@ -16,13 +25,21 @@
...
-    """Displays the homepage."""
+    """License lookup form render"""
+    # Check for badge query params and load if it exists.
+    license = request.args.get('license')
+    if license:
+        html = dataset.license_lookup(license)
+        return render_template("index.html", **LICENSE_CONTEXT, entity_html=html)
+
+    # Return the basic lookup form if not
     return render_template("index.html", **LICENSE_CONTEXT)
```
Comment (AetherUnbound) [inline src/app.py]: For each of these cases, I'd suggest the following change for simplicity:

```suggestion
    license = request.args.get('license')
    html = ""
    if license:
        html = dataset.license_lookup(license)

    return render_template("index.html", **LICENSE_CONTEXT, entity_html=html)
```

This only hydrates the extra HTML if it's available, and still renders everything well when it isn't.

**Review by AetherUnbound — COMMENTED** [review]
Dang, this is awesome! You're right, way easier than I was expecting. I checked that the old links are still preserved and they are! One comment about formatting, then I think this is good to go :rocket:

File: src/app.py (hunk 2 above)
Comment (mpuckett159) [reply src/app.py]: Oh this makes sense, I didn't think about that. I'll make those changes.

**Review by AetherUnbound — APPROVED** [review]
Way to go :sunglasses:

**Pull request merged**